Лабораторная работа 4

kozlovkirilI opened the pull request.

kozlovkirilI committed:
Лабораторная работа 4

andreiivanov-wa commented on Лабораторная работа 4/task1.py:
```diff
@@ -0,0 +1,109 @@
+        self.radius = radius
+
+    @property
+    def area(self) -> float:
```
почему это свойство, а не метод?

andreiivanov-wa commented on Лабораторная работа 4/task1.py:
```diff
@@ -0,0 +1,109 @@
+    rect = Rectangle([1, 6], 3, 4)
+    print(rect)
+    print(repr(rect))
+    print(rect.area)
```
у вас в коде используется только наследование атрибутов, но не методов
Например, можно было предложить реализовать модель четырехугольников. Наследуемым методом является периметр для базового четырехугольника, вычисляется как сумма всех сторон для любого четырехугольника одинаково. Перегружаемым методом может быть площадь -- вычисляется по-разному для разных четырехугольников.

kozlovkirilI committed:
Update task1.py

Изменил класс Figure на Tetragon для четырехугольников, Circle заменил на Square. Теперь наследуется метод perimeter и перегружается метод area.
